### PR TITLE
feat: support json content in requests

### PR DESCRIPTION
**Pull request by shrutiburman**
Fixes [DII-1005](https://issues.corp.twilio.com/browse/DII-1005)

This PR: 
Supports sending json payload in Http Requests. 

### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [ ] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [ ] I have read the [Contribution Guidelines](https://github.com/twilio/twilio-java/blob/main/CONTRIBUTING.md) and my PR follows them
- [ ] I have titled the PR appropriately
- [ ] I have updated my branch with the main branch
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation about the functionality in the appropriate .md file
- [ ] I have added inline documentation to the code I modified

If you have questions, please file a [support ticket](https://twilio.com/help/contact), or create a GitHub Issue in this repository.

### CODE REVIEW

File: src/main/java/com/twilio/http/ValidationClient.java [hunk 1]
```diff
@@ -104,7 +107,14 @@ public Response makeRequest(Request request) {
 
         HttpMethod method = request.getMethod();
         if (method == HttpMethod.POST) {
-            builder.addHeader(HttpHeaders.CONTENT_TYPE, "application/x-www-form-urlencoded");
+            if(request.getContentType() != null && request.getContentType().equals("APPLICATION_JSON")){
+                HttpEntity entity = new StringEntity(request.encodeFormBody(), ContentType.APPLICATION_JSON);
```
Comment (shrutiburman) [inline src/main/java/com/twilio/http/ValidationClient.java:111]: [self] : We can also include a json_payload body parameter in request to be sent here in entity, in place of `request.encodeFormBody()` . @sbansla thoughts ?

File: src/main/java/com/twilio/http/ValidationClient.java [hunk 2]
```diff
@@ -104,7 +107,14 @@ public Response makeRequest(Request request) {
 
         HttpMethod method = request.getMethod();
         if (method == HttpMethod.POST) {
-            builder.addHeader(HttpHeaders.CONTENT_TYPE, "application/x-www-form-urlencoded");
+            if(request.getContentType() != null && request.getContentType().equals("APPLICATION_JSON")){
```
Comment (allantodd) [inline src/main/java/com/twilio/http/ValidationClient.java:110]: Why is the content type the string literal "APPLICATION_JSON".  Shouldn't it be "application/json"?

File: src/test/java/com/twilio/http/ValidationClientTest.java [hunk 3]
```diff
@@ -46,6 +47,27 @@ private void exerciseHttpMethod(final HttpMethod httpMethod) throws Exception {
...
+        final String path = "/example123";
+        final HttpUrl url = server.url(path);
+        final ValidationClient client = new ValidationClient("dummy-sid1", "dummy-sid2", "dummy-signing-key", keyPair.getPrivate());
+        final Request request = new Request(httpMethod, url.url().toString());
+        request.setContentType("APPLICATION_JSON");
+        final Response response = client.makeRequest(request);
+        assertEquals(200, response.getStatusCode());
+        final RecordedRequest recordedRequest = server.takeRequest();
+        assertEquals(httpMethod.name(), recordedRequest.getMethod());
+        assertEquals("APPLICATION_JSON", recordedRequest.getHeader("Content-Type"));
```
Comment (allantodd) [inline src/test/java/com/twilio/http/ValidationClientTest.java:66]: As above, why "APPLICATION_JSON"?  Maybe ContentType.APPLICATION_JSON ?

File: src/main/java/com/twilio/http/ValidationClient.java (hunk 2 above)
Comment (shrutiburman) [reply src/main/java/com/twilio/http/ValidationClient.java:110]: Yes

**Review by sbansla — APPROVED** [review]

**Pull request merged**